src/layout/monitor.rs

```diff
@@ -490,9 +490,16 @@ impl<W: LayoutElement> Monitor<W> {
         }
         let new_id = self.workspaces[new_idx].id();
 
-        let activate = window.map_or(true, |win| {
-            self.active_window().map(|win| win.id()) == Some(win)
-        });
+        // Whether to activate the window based on focus parameter and whether the window
+        // is currently focused.
+        let activate = if focus {
```

Comment:
Could you replace `focus: bool` with `activate: ActivateWindow` then use `map_smart()` for this check? Pass `Smart` for the current behavior, and `No` for new behavior